### PR TITLE
9.nc news

### PR DESCRIPTION
**Pull request by OlliePatty**

### CODE REVIEW

File: src/components/Home.jsx [hunk 1]
```diff
@@ -23,7 +25,11 @@ export default function Home() {
   loading ? <p className='loading'>Loading</p> : (
     <main className='article-list'>
       {articles.map((article)=>{
```
Comment (Beth-McCormack) [inline src/components/Home.jsx]: Filtering by topic should not be done on the front end, but by utilising back end queries. Have a look at the NC notes on axios for some pointers on this🙂

**Review by Beth-McCormack — CHANGES_REQUESTED** [review]
I've left a comment in your code, please have a look and submit another PR when this is done 🙂

File: src/components/ArticlesList.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import ArticleCard from './ArticleCard'
+import Error from './Error'
+
+export default function Home() {
+  const { topic } = useParams()
+  const [ articles, setArticles ] = useState([])
+  const [ loading, setLoading ] = useState(true)
+  const [ error, setError ] = useState(null)
+
+  useEffect(()=>{
```
Comment (NRMSMITH) [inline src/components/ArticlesList.jsx:13]: I would suggest that you should reset your error to null here :)

File: src/components/Topics.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect } from 'react'
+import { getAllTopics } from '../utils'
+import { Link } from 'react-router-dom'
+
+export default function Topics() {
+    const [ loading, setLoading ] = useState(true)
+    const [ topics, setTopics ] = useState([])
+
+    useEffect(()=>{
```
Comment (NRMSMITH) [inline src/components/Topics.jsx:11]: where is set Error established here?

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Have a look over your set Errors :) (good work on making the api do the work!)

File: src/components/Topics.jsx [hunk 4]
```diff
@@ -17,7 +19,8 @@ export default function Topics() {
         })
       }, [])
 
-  return loading ? <p className='loading'>loading</p> : (
+  return error ? <Error error={error}/> :
+  loading ? <p className='loading'>Loading</p> : (
```
Comment (xViolaine) [inline src/components/Topics.jsx:23]: Good job taking the feedback given to you on board! Just a small note; nested ternaries get confusing quickly. This is still mostly alright, but anything more than that, and I would recommend a different way to go about this :D

**Review by xViolaine — APPROVED** [review]

**Pull request merged**